src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m

```diff
@@ -277,6 +283,7 @@ CVReturn displayLinkCallback(CVDisplayLinkRef displayLink, const CVTimeStamp* no
     MTLContext * ctx = layer.ctx;
     if (layer == NULL || ctx == NULL) {
         J2dTraceLn(J2D_TRACE_VERBOSE, "MTLLayer_blit : Layer or Context is null");
+        [layer stopDisplayLink];
```

Comment:
What happens if we exit before this "blitTexture" is called? Do not we need to stop(w/o possibility to restart it) that thread on toolkit shutdown or something like this?

Reply:
At this point if we exit, we just return and unlock MTLRenderQueue for backbuffer rendering. And when it is done we again start CVDisplayLink in MTLRenderQueue which in turn calls setNeedsDisplay to get callback to MTLLayer.display().

This scenario is same as returning from CGLLayer.blitTexture() when textureID is 0. Appkit thread will be running in the background and we dont stop it.

Reply:
No, I meant if we dispose the frame/exist from the app/etc before "Java_sun_java2d_metal_MTLLayer_blitTexture" method is executed but after we "startDisplayLink", when we will call the stopDisplayLink in this case?

Reply:
In that case it will hit MTLLayer.dealloc() where we stop and release CVDisplayLink.

Reply:
@mrserb Please let me know if i can resolve this conversation.

Reply:
ok, looks fine